src/seedu/addressbook/commands/Command.java

```diff
@@ -37,6 +37,14 @@ public static String getMessageForPersonListShownSummary(List<? extends ReadOnly
     }
 
     /**
+     *
+     * @return if the (child) class will mutate the data
+     */
+    public boolean isMutating() {
+        throw new UnsupportedOperationException("This method should be implement in child classes");
+    }
```

Comment:
You probably made it this way because abstract classes is not part of this LO. But as you know, this can be `abstract`.